### PR TITLE
Rwa-1437-4: available task only search 

### PR DESCRIPTION
**Pull request by rasikasri**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskSearchControllerTest.java [hunk 1]
```diff
@@ -492,4 +497,166 @@ public void should_return_200_with_tasks_sorted_on_next_hearing_date_asc() {
...
+        common.setupWAOrganisationalRoleAssignment(granularPermissionCaseworkerCredentials.getHeaders(),
+                                                   "tribunal-caseworker");
+
+        /**
+         *                      processApplication      followUpOverdue     inspectAppeal
+         * tribunal-caseworker  READ,OWN,CLAIM          READ                READ,OWN,MANAGE
+         * general search       v                       v                   v
+         * available task       v                       x                   x
+         * all work             x                       x                   v
+         */
```
Comment (raghera) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskSearchControllerTest.java:544]: 👍

File: src/main/java/net/hmcts/taskperf/service/TaskSearch.java [hunk 2]
```diff
@@ -195,25 +195,16 @@ private static Set<String> buildPermissions(ClientFilter clientFilter)
...
-			//       makes the query complex - we should probably not expose
-			//       the ability to request specific permissions through the API,
-			//       but flags/properties for particular types of query, which
-			//       we then interpret to determine what permissions we should
-			//       search for.  (There are other reasons for this, too - we
-			//       don't want to have to index on all the permissions, just
-			//       the ones we need for the search scenarios we have to support.
-			permissions.add("o");
-//			permissions.add("x");
+			permissions.add("a");
```
Comment (raghera) [inline src/main/java/net/hmcts/taskperf/service/TaskSearch.java:198]: Maybe put a comment to explain what `a` refers to i.e. A DB view that represents 2 permissions?

**Review by raghera — APPROVED** [review]
Nicely done.  
Just a suggestion around comments but that's more for info.

File: src/main/java/net/hmcts/taskperf/service/TaskSearch.java (hunk 2 above)
Comment (rasikasri) [reply src/main/java/net/hmcts/taskperf/service/TaskSearch.java:199]: Done

**Pull request merged**